### PR TITLE
Improve dmoj.utils.*

### PR DESCRIPTION
**Pull request by quantum5**
Most just adds `mypy` and switch to f-strings. Also start using `signal.strsignal` instead of `ctypes`.

### CODE REVIEW

File: dmoj/utils/os_ext.py [hunk 1]
```diff
@@ -2,40 +2,48 @@
...
+from typing import Optional
 
 from dmoj.utils.unicode import utf8bytes
 
 OOM_SCORE_ADJ_MAX = 1000
 OOM_SCORE_ADJ_MIN = -1000
 
 
-def oom_score_adj(score, to=None):
+def oom_score_adj(score, to: Optional[int] = None) -> None:
```
Comment (kiritofeng) [inline dmoj/utils/os_ext.py:13]: Should we annotate `score` as well?

**Review by kiritofeng — COMMENTED** [review]
Otherwise LGTM

File: dmoj/utils/os_ext.py (hunk 1 above)
Comment (quantum5) [reply dmoj/utils/os_ext.py]: Yes.

**Review by kiritofeng — APPROVED** [review]
LGTM

**Pull request merged**